pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -62,7 +62,14 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 		return fmt.Errorf("couldn't list Deployments for DeploymentConfig %s: %v", deployutil.LabelForDeploymentConfig(config), err)
 	}
 	var inflightDeployment *kapi.ReplicationController
+	latestDeploymentExists := false
 	for _, deployment := range existingDeployments.Items {
+		// check if this is the latest deployment
+		// we'll return after we've dealt with the multiple-active-deployments case
+		if deployutil.DeploymentVersionFor(&deployment) == config.LatestVersion {
+			latestDeploymentExists = true
```

Comment:
```
// we'll return after we've dealt with the multiple-active-deployments case
```
Did try to cover that with a comment - we would like to identify and clean up concurrently active deployments